Earthfile

```diff
@@ -99,57 +59,74 @@ docker:
         SAVE IMAGE --push $image:$tag
     END
 
-deps:
-    FROM +source
-    COPY +build/partner-chains-node .
-    # calculate libary deps
-    RUN ldd partner-chains-node \
-        | awk 'NF == 4 { system("echo " $3) }' \
-        | tar -czf deps.tgz --files-from=-
-    SAVE ARTIFACT deps.tgz deps
-
-mock:
-  FROM +setup
-  ARG CRATES=$(tomlq -r '.workspace.members[]' Cargo.toml)
-  ARG SRCS=$(tomlq -r '.workspace.members[] + "/src"' Cargo.toml)
-  ARG LIBS=$(tomlq -r '.workspace.dependencies[]|select(type == "object" and has("path")).path + "/src/lib.rs"' Cargo.toml)
-  FOR crate IN $CRATES
-    COPY --if-exists $crate/Cargo.toml $crate/Cargo.toml
+test:
+  FROM +build
+  LET WASM_BUILD_STD=0
+  DO github.com/earthly/lib:3.0.3+INSTALL_DIND
+  CACHE --sharing shared --id cargo $CARGO_HOME
+  RUN cargo test --no-run --locked --profile=$PROFILE --features=$FEATURES,runtime-benchmarks
+  WITH DOCKER
+    RUN cargo test --locked --profile=$PROFILE --features=$FEATURES,runtime-benchmarks
```

Comment:
I know it's not new here, but why do we have to `cargo test --no-run` before running the actual tests?

Reply:
Just another layer optmization, to split building the test executables from actually running them.

In general in docer setups (not just earthly), it is typical and good practice to split steps into descreet units to maximize the layer caches effectiveness.